app/src/marketplace/containers/ProductPage/PurchaseModal.jsx

```diff
@@ -116,6 +114,17 @@ export const PurchaseDialog = ({ productId, api }: Props) => {
             })
     }, [dispatch, loadEthIdentities, loadContractProduct, productId, isMounted])
 
+    useEffect(() => {
```

Comment:
moved balance querying to here, it only needs to be done once (before it was everytime currency changed)